packages/iov-bns/src/encode.ts

```diff
@@ -498,15 +497,15 @@ export function buildMsg(tx: UnsignedTransaction): codecImpl.bnsd.ITx {
 export function buildUnsignedTx(tx: UnsignedTransaction): codecImpl.bnsd.ITx {
   const msg = buildMsg(tx);
 
-  let feePayerAddressBytes: Uint8Array;
-  if (tx.fee && tx.fee.payer) {
-    feePayerAddressBytes = decodeBnsAddress(tx.fee.payer).data;
-  } else if (isMultisignatureTx(tx)) {
-    const firstContract = tx.multisig.find(() => true);
-    if (firstContract === undefined) throw new Error("Empty multisig arrays are currently unsupported");
-    feePayerAddressBytes = conditionToWeaveAddress(buildMultisignatureCondition(firstContract));
-  } else {
-    feePayerAddressBytes = decodeBnsAddress(identityToAddress(tx.creator)).data;
+  let feePayerAddressBytes: Uint8Array = new Uint8Array();
```

Comment:
We add a little bit of type strangeness here by requiring to set `feePayerAddressBytes` for the case `tx.fee && tx.fee.tokens` is false. However, we only need the variable for `tx.fee && tx.fee.tokens` is true (see below). Can we rewrite this bit in a way that only sets `const feePayerAddressBytes` for `tx.fee && tx.fee.tokens`?